### PR TITLE
Feature/changes image gallery

### PR DESCRIPTION
**Pull request by axelprog**
new examples and functions for the gallery

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,15 +38,17 @@ var components = {
 var redaxtor = new Redaxtor({
     pieces: {
         components: components,
-        getURL: "api/pieces/get",//will be overwritten by data-get-url
-        saveURL: "api/pieces/save"//will be overwritten by data-save-url
-    }
+        getURL: "api/pieces/get", //will be overwritten by data-get-url
```
Comment (AndrewKirkovski) [inline README.md]: Replace this sample with `api` one, getUrl and saveUrl are deprecated

File: src/reducers/pieces.js [hunk 2]
```diff
@@ -3,11 +3,24 @@ import C from '../constants'
...
+                changed: !(action.notChanged || (action.piece.data.html === piece.data.html)) || piece.changed
+            };
         case C.PIECE_RESET:
             return {...piece, changed: false};
         case C.PIECE_REMOVE:
             return {...piece, destroy: true};
+        case C.PIECE_SET_DATA:
+            //check is initiated
+            if(!piece.fetched){
+                console.error(`Piece was not initialized before use setDate function. Piece id: ${piece.id}`);
```
Comment (AndrewKirkovski) [inline src/reducers/pieces.js:17]: Additionally add a check in setData public method and produce an error with `throw new Error` there if there is no piece data or no 'byId' at all

File: src/reducers/pieces.js [hunk 3]
```diff
@@ -88,6 +118,13 @@ const pieces = (pieces = piecesDefault, action) => {
                 byId: {...pieces.byId, [action.id]: piece(pieces.byId[action.id], action)}
             };
         default:
+            /*init default empty value*/
```
Comment (AndrewKirkovski) [inline src/reducers/pieces.js]: Drop this. Default action in redux shouldn't do anything. Instead use `pieces.byId && pieces.byId[...]` checks where needed

Comment (axelprog) [reply src/reducers/pieces.js]: all this issues

**Pull request merged**